### PR TITLE
Square Root exercise

### LINKED ISSUE #633: Adding the square-root exercise
Hi!

I'd like to add the square-root exercise to the track.

### PR DESCRIPTION
**Pull request by golanor**
Hi,

This is an attempt at adding the square-root exercise to Julia. This is my first PR, so please let me know (and be forgiving) if I missed something. Should close #633 

Thanks!

### CODE REVIEW

File: exercises/practice/square-root/square-root.jl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+function square_root(n::Int64; tol)
```
Comment (cmcaine) [inline exercises/practice/square-root/square-root.jl:1]: Do we need to provide a tolerance parameter? The built-in sqrt doesn't use one.

If we do provide a tolerance parameter then I think we should specify if it is an absolute tolerance (atol) or relative tolerance (rtol) to match base Julia's naming patterns.

File: exercises/practice/square-root/.meta/example.jl [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    x = initial_guess(n)
+    while abs(x^2 - n) > tol
+        x = 0.5 * (x + n / x)
+    end
+    round(x; digits=digs)
+end
+
+
+function initial_guess(n::Int64)
+    m = Int(floor(log10(n) / 2))
```
Comment (cmcaine) [inline exercises/practice/square-root/.meta/example.jl:15]: This isn't important, but this pattern is a bit more idiomatic, I think, and better matches your line 2.

```suggestion
    m = floor(Int, log10(n) / 2)
```

File: exercises/practice/square-root/.meta/config.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "blurb": "Given a natural radicand, return its square root.",
```
Comment (cmcaine) [inline exercises/practice/square-root/.meta/config.json:2]: With less jargon?

```suggestion
  "blurb": "Calculate a positive integer's square root",
```

File: exercises/practice/square-root/square-root.jl [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+    
+end
```
Comment (cmcaine) [inline exercises/practice/square-root/square-root.jl:3]: Stubs should end with a blank line, I think, to match unix expectations.

```suggestion
end

```

**Review by cmcaine — COMMENTED** [review]
Thanks!

File: exercises/practice/square-root/square-root.jl (hunk 1 above)
Comment (cmcaine) [reply exercises/practice/square-root/square-root.jl:1]: Here's an implementation that will converge on the exact answers for integer square numbers:

```julia
function mysqrt2(S)
    x = S
    while (S/x + x) / 2 ≠ x
        x = (S/x + x) / 2
    end
    return x
end


findall(sqrt(n^2) ≠ mysqrt2(n^2) for n in 1:1000)
```

Comment (golanor) [reply exercises/practice/square-root/square-root.jl:1]: I thought having a keyword argument would be a nice practice that is usually not done in these exercises, but that's not a must.

Comment (cmcaine) [inline exercises/practice/square-root/square-root.jl:1]:
```suggestion
function square_root(n)
```

File: exercises/practice/square-root/runtests.jl [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    @testset "Domain error for negative number" begin
+        @test_throws DomainError square_root(-65536)
+    end
+end
+
+
+@testset "Random tests" begin
+    for s in rand(10, 1:1000000)
+        @test square_root(s^2) == s
+    end
```
Comment (cmcaine) [inline exercises/practice/square-root/runtests.jl:34]:
```suggestion
    for _ in 1:1000
        s = rand(1:isqrt(typemax(Int64)))
        @test square_root(s^2) == s
    end
```

File: exercises/practice/square-root/square-root.jl (hunk 1 above)
Comment (cmcaine) [reply exercises/practice/square-root/square-root.jl:1]: Cool, well I think it's better to not have it in this exercise. Thanks for removing it!

**Review by cmcaine — APPROVED** [review]

File: exercises/practice/square-root/.meta/example.jl [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    end
+    x
+end
+
+
+function initial_guess(n::Int64)
+    m = floor(Int, log10(n) / 2)
+    a = n ÷ (10 ^ m)
+    (a < 10 ? (0.89 + 0.28*a) : (2.8 + 0.089*a)) * 10^m
+end
```
Comment (cmcaine) [inline exercises/practice/square-root/.meta/example.jl:17]: This equation is a little messed up, btw:

```suggestion
function initial_guess(n::Int64)
    m = floor(Int, log10(n))
    a = n / (10 ^ m)
    (a < 10 ? (0.89 + 0.28*a) : (2.8 + 0.089*a)) * 10^(m ÷ 2)
end
```

**Pull request merged**